Feat(web-react): Introduce UNSTABLE_Avatar component #DS-1324 (#1477)
lmc-eu/spirit-design-system: packages/web-react/src/components/UNSTABLE_Avatar/README.md

# UNSTABLE Avatar

⚠️ This component is UNSTABLE. It may significantly change at any point in the future.
Please use it with caution.

The `UNSTABLE_Avatar` component is used to display a user's profile picture or initials.

```jsx
import { UNSTABLE_Avatar } from '@lmc-eu/spirit-web-react';

> <UNSTABLE_Avatar aria-label="Profile of Jiří Bárta">

Reviewer: ```suggestion
<UNSTABLE_Avatar ariaLabel="Profile of Jiří Bárta">
```

>   <span aria-hidden="true">JB</span>

Reviewer: ```suggestion
  <span ariaHidden="true">JB</span>
```

</UNSTABLE_Avatar>;
```

## Square

> Add `isSquare` prop to make the avatar a square.

Reviewer 2: This could be set like `shape=[ circle (default) | square ]`.

I can see there is probably an analogy with the `Button` component and its `isSquare` prop, although this is not quite the same situation…

I'm OK with both `square` and `isSquare` variants. Just if you decide for the `square` enum prop, CSS should be changed too.

https://github.com/lmc-eu/spirit-design-system/pull/1469/files#r1647648629

Reviewer 3: I'll leave it as it is for now, it is an unstable component, so we can change it later if we decide to.

```jsx

> <UNSTABLE_Avatar isSquare aria-label="Profile of Jiří Bárta">

Reviewer: ```suggestion
<UNSTABLE_Avatar isSquare ariaLabel="Profile of Jiří Bárta">
```

>   <span aria-hidden="true">JB</span>

Reviewer: ```suggestion
  <span ariaHidden="true">JB</span>
```

Reviewer: And everywhere else.

</UNSTABLE_Avatar>
```

## Sizes

> The Avatar component supports `xsmall`, `small`, `medium`, `large`, and `xlarge` sizes.

Reviewer: Please, use a dictionary and mention it.

```jsx
<UNSTABLE_Avatar aria-label="Profile of Jiří Bárta" size="xsmall">
  <span aria-hidden="true">JB</span>
</UNSTABLE_Avatar>
<UNSTABLE_Avatar aria-label="Profile of Jiří Bárta" size="small">
  <span aria-hidden="true">JB</span>
</UNSTABLE_Avatar>
<UNSTABLE_Avatar aria-label="Profile of Jiří Bárta" size="medium">
  <span aria-hidden="true">JB</span>
</UNSTABLE_Avatar>
<UNSTABLE_Avatar aria-label="Profile of Jiří Bárta" size="large">
  <span aria-hidden="true">JB</span>
</UNSTABLE_Avatar>
<UNSTABLE_Avatar aria-label="Profile of Jiří Bárta" size="xlarge">
  <span aria-hidden="true">JB</span>
</UNSTABLE_Avatar>
```

## Content

The content of the `UNSTABLE_Avatar` component can be an image, an icon, or a text string.

### Icon

Add an Icon with correct size.

```jsx
<UNSTABLE_Avatar size="xsmall" aria-label="Profile of Jiří Bárta">
  <Icon name="profile" boxSize="16" />
</UNSTABLE_Avatar>
<UNSTABLE_Avatar aria-label="Profile of Jiří Bárta">
  <Icon name="profile" boxSize="24" />
</UNSTABLE_Avatar>
```

> ℹ️ Don't forget to add the `aria-label` attribute for accessible title.

Reviewer: ```suggestion
ℹ️ Don't forget to add the `ariaLabel` prop for accessible title.
```

Reviewer: Oh, shit https://legacy.reactjs.org/docs/accessibility.html#wai-aria
